### PR TITLE
Fix merge

### PR DESCRIPTION
**Pull request by acestronautical**
Fix merging to use copypages instead of embedPDF. This resolves a margin issue where margins were being used from the source and destination PDFs in a weird way. 

Replace hopding/pdf-lib with the maintained cantoo-scribe/pdf-lib to which we have an upstreamed performance fix for copypages https://github.com/cantoo-scribe/pdf-lib/pull/48

Perform an npm audit fix and npm update

improve upload performance by pulling some variables out of the page rotation loop

### CODE REVIEW

**Comment by momijizukamori** [comment]
Can you do a quick filesize check? I feel like `copyPages` had more overhead and that's why I changed it....

**Comment by acestronautical** [comment]
Not sure why the import is failing after switching to the cantoo version? It works fine on my local hosted version?

File size seems the same for me, but I'm using a mostly test PDF

File: package.json [hunk 1]
```diff
@@ -17,15 +17,16 @@
   "license": "MPL",
   "devDependencies": {
     "eslint": "^8.56.0",
+    "jsdom": "^24.0.0",
```
Comment (acestronautical) [inline package.json:20]: This is needed by vite.config.test, it's not really a change just documenting a dependency that happened to resolve before. This way we don't have to get lucky.

File: src/book.js [hunk 2]
```diff
@@ -128,16 +128,6 @@ export class Book {
...
-          const cropBox = page.getCropBox();
-          const bleedBox = page.getBleedBox();
-          const trimBox = page.getTrimBox();
-          const artBox = page.getArtBox();
-          console.log(
-            `\n\tCropBox [${cropBox}]`,
-            `\n\tBleedBox [${bleedBox}]`,
-            `\n\tTrimBox [${trimBox}]`,
-            `\n\tArtBox [${artBox}]`
-          );
```
Comment (acestronautical) [inline src/book.js:140]: Unfortunately this console logs are actually a bit expensive. We could comment them out or add a debug flag somewhere for verbose output, but we shouldn't be shipping this.

File: src/book.js [hunk 3]
```diff
@@ -185,18 +175,24 @@ export class Book {
...
     [this.managedDoc, pages] = await embedPagesInNewPdf(this.currentdoc);
 
+    const rotateNone = this.source_rotation == 'none';
+
+    const rotate90cw =
+      this.source_rotation == '90cw' ||
+      (this.source_rotation == 'out_binding' && pages.length % 2 == 0) ||
+      (this.source_rotation == 'in_binding' && pages.length % 2 == 1);
+
+    const rotate90ccw =
```
Comment (acestronautical) [inline src/book.js:185]: Pulling these out removes a lot of unnecessary string comparisons

File: src/book.js [hunk 4]
```diff
@@ -185,38 +175,40 @@ export class Book {
     let pages;
     [this.managedDoc, pages] = await embedPagesInNewPdf(this.currentdoc);
 
+    const isNone = this.source_rotation == 'none';
+    const is90cw = this.source_rotation == '90cw';
+    const is90ccw = this.source_rotation == '90ccw';
+    const isInBinding = this.source_rotation == 'in_binding';
+    const isOutBinding = this.source_rotation == 'out_binding';
```
Comment (acestronautical) [inline src/book.js:182]: This is a little verbose but it saves us a lot string comparisons in the loop

**Review by momijizukamori — APPROVED** [review]
Looks good! Thank you for the notes on some of the changes :) I bet jsdom was a dep of the original library version and isn't on the new one or something, heh.

File: src/book.js (hunk 2 above)
Comment (sithel) [reply src/book.js:140]: 👍 I can't say I'm aware ATM of any logs I value... I think I tend to add them when I need them and shortly after but then... the problem's always somewhere else next time...

**Pull request merged**